onnxruntime/core/common/type_list.h

```diff
@@ -9,4 +9,9 @@ namespace onnxruntime {
 template <typename... T>
 struct TypeList {};
 
-}
+}  // namespace onnxruntime
+
+// type list type containing the given types
+// Note: this is useful for passing TypeLists to macros which don't accept the
+//       comma-separated template arguments
+#define ORT_TYPE_LIST(...) ::onnxruntime::TypeList<__VA_ARGS__>
```

Comment:
>(... [](start = 21, length = 4)

Is empty typelist OK?

Reply:
yes, that's also valid

---
In reply to: [572282034](https://github.com/microsoft/onnxruntime/pull/6594#discussion_r572282034) [](ancestors = 572282034)